### PR TITLE
Closes #71 | Create dataset loader for MASSIVE

### LINKED ISSUE #71: Create dataset loader for MASSIVE
Dataloader name: `massive/massive.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?massive

|     Dataset| massive |
|-------------|---|
| Description | MASSIVE dataset—Multilingual Amazon Slu resource package (SLURP) for Slot-filling, Intent classification, and Virtual assistant Evaluation. MASSIVE contains 1M realistic, parallel, labeled virtual assistant utterances spanning 18 domains, 60 intents, and 55 slots. MASSIVE was created by tasking professional translators to localize the English-only SLURP dataset into 50 typologically diverse languages, including 8 native languages and 2 other languages mostly spoken in Southeast Asia. |
| Subsets | id-ID, jv-ID, km-KH, ms-MY, my-MM, th-TH, tl-PH, vi-VN |
| Languages | ind, jav, khm, zlm, mya, tha, tgl, vie, cmn, tam |
| Tasks | Slot Filling, Intent Classification |
| License | Creative Commons Attribution 4.0 (cc-by-4.0) |
| Homepage | https://github.com/alexa/massive |
| HF URL | https://huggingface.co/datasets/qanastek/MASSIVE |
| Paper URL | https://aclanthology.org/2023.acl-long.235/ |

### PR DESCRIPTION
**Pull request by yana-xuyan**
Closes #71.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/massive/massive.py
Comment (gentaiscool) [inline seacrowd/sea_datasets/massive/massive.py:228]: can you modify the class name to MASSIVEDataset for consistency to other datasets?

File: seacrowd/sea_datasets/massive/massive.py
Comment (gentaiscool) [inline seacrowd/sea_datasets/massive/massive.py:358]: refactor the function name with snake case.

**Review by gentaiscool — CHANGES_REQUESTED** [review]
I was wondering if we should split the slot filling and intent detection dataset. tag @holylovenia @SamuelCahyawijaya @sabilmakbar

**Comment by yana-xuyan** [comment]
Hi Genta, sure. For separating the slot filling anc intent detection dataset, I can modify if needed. :)

Best regards,
Yan

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Genta Indra Winata ***@***.***>
发送时间: Wednesday, January 3, 2024 10:50:26 AM
收件人: SEACrowd/seacrowd-datahub ***@***.***>
抄送: Yan XU ***@***.***>; Author ***@***.***>
主题: Re: [SEACrowd/seacrowd-datahub] Closes #71 | Create dataset loader for MASSIVE (PR #196)


@gentaiscool requested changes on this pull request.

I was wondering if we should split the slot filling and intent detection dataset. tag @holylovenia<https://github.com/holylovenia> @SamuelCahyawijaya<https://github.com/SamuelCahyawijaya> @sabilmakbar<https://github.com/sabilmakbar>

________________________________

In seacrowd/sea_datasets/massive/massive.py<https://github.com/SEACrowd/seacrowd-datahub/pull/196#discussion_r1440017954>:

> +    'I-coffee_type', 'B-transport_agency', 'B-podcast_descriptor', 'I-playlist_name', 'B-media_type', 'B-song_name', 'I-music_descriptor',
+    'I-song_name', 'B-event_name', 'I-timeofday', 'B-alarm_type', 'B-cooking_type', 'I-business_name', 'I-color_type', 'B-podcast_name',
+    'I-personal_info', 'B-weather_descriptor', 'I-list_name', 'B-transport_descriptor', 'I-game_type', 'I-date', 'B-place_name', 'B-color_type',
+    'B-game_name', 'I-artist_name', 'I-drink_type', 'B-business_name', 'B-timeofday', 'B-sport_type', 'I-player_setting', 'I-transport_agency',
+    'B-game_type', 'B-player_setting', 'I-music_album', 'I-event_name', 'I-general_frequency', 'I-podcast_name', 'I-cooking_type', 'I-radio_name',
+    'I-joke_type', 'I-meal_type', 'I-transport_type', 'B-joke_type', 'B-time', 'B-order_type', 'B-business_type', 'B-general_frequency',
+    'I-food_type', 'I-time_zone', 'B-currency_name', 'B-time_zone', 'B-ingredient', 'B-house_place', 'B-audiobook_name', 'I-ingredient',
+    'I-media_type', 'I-news_topic', 'B-music_genre', 'I-definition_word', 'B-list_name', 'B-playlist_name', 'B-email_address', 'I-currency_name',
+    'I-movie_name', 'I-device_type', 'I-weather_descriptor', 'B-audiobook_author', 'I-audiobook_author', 'I-app_name', 'I-order_type',
+    'I-transport_name', 'B-radio_name', 'I-business_type', 'B-definition_word', 'B-artist_name', 'I-movie_type', 'B-transport_name', 'I-email_folder',
+    'B-music_album', 'I-house_place', 'I-music_genre', 'B-drink_type', 'I-alarm_type', 'B-music_descriptor', 'B-news_topic', 'B-meal_type',
+    'I-transport_descriptor', 'I-email_address', 'I-change_amount', 'B-device_type', 'B-transport_type', 'B-relation', 'I-sport_type', 'B-personal_info'
+]
+
+
+class MASSIVE(datasets.GeneratorBasedBuilder):


can you modify the class name to MASSIVEDataset for consistency to other datasets?

________________________________

In seacrowd/sea_datasets/massive/massive.py<https://github.com/SEACrowd/seacrowd-datahub/pull/196#discussion_r1440018120>:

> +                    "split": "dev",
+                    "lang": self.config.name,
+                }
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "files": dl_manager.iter_archive(archive),
+                    "split": "test",
+                    "lang": self.config.name,
+                }
+            ),
+        ]
+
+
+    def _getBioFormat(self, text):


refactor the function name with snake case.

―

**Comment by yana-xuyan** [comment]
Hi @gentaiscool, I have modified the code accordingly. Do you think we need to split the slot filling and intent detection dataset btw?

**Comment by holylovenia** [comment]
> Hi @gentaiscool, I have modified the code accordingly. Do you think we need to split the slot filling and intent detection dataset btw?

Hi @yana-xuyan, can we separate the tasks into different `seacrowd` subsets (similar to #235)?

**Comment by yana-xuyan** [comment]
Okie. Thank you Holy!

**Review by SamuelCahyawijaya — APPROVED** [review]
Hi @yana-xuyan, thank you for contributing. Tested and LGTM! Let's wait for the review from @gentaiscool 👍🏻

**Review by gentaiscool — APPROVED** [review]
thanks Xuyan for the contribution. Lgtm.

**Pull request merged**
